File: src/types/reliability.ts
import type { DiagnosticItem } from "@/types/diagnostics";
import type { PersistentTunnelHealthReport, ProjectPersistentHostname, ProjectPersistentTunnelState } from "@/types/persistent-tunnel";
import type { Project } from "@/types/project";
import type { ServiceState } from "@/types/service";
import type { ProjectTunnelState } from "@/types/tunnel";

export type ReliabilityLayer =
  | "project"
  | "runtime"
  | "config"
  | "service"
  | "dns"
  | "tunnel"
  | "workspace";

export type ReliabilityStatus = "ok" | "warning" | "error";

export type ReliabilityAction =
  | "provisionProject"
  | "publishPersistentDomain"
  | "startProjectRuntime"
  | "restoreAppMetadata";

export type RepairWorkflow = "project" | "tunnel" | "runtimeLinks";

export interface ReliabilityCheck {
  code: string;
  layer: ReliabilityLayer;
  status: ReliabilityStatus;
  blocking: boolean;
  title: string;
  message: string;
  suggestion?: string | null;
}

export interface ActionPreflightReport {
  action: ReliabilityAction;
  projectId?: string | null;
  ready: boolean;
  summary: string;
  checks: ReliabilityCheck[];
  generatedAt: string;
}

export interface RepairWorkflowInfo {
  workflow: RepairWorkflow;
  title: string;
  summary: string;
  touches: string[];
}

export interface RepairExecutionResult {
  workflow: RepairWorkflow;
  success: true;
  message: string;
  touchedLayers: ReliabilityLayer[];
  generatedAt: string;
}

export interface InspectorConfigSnapshot {
  serverType: Project["serverType"];
  outputPath: string;
  preview?: string | null;
  localDomainAliasPresent: boolean;
  persistentAliasPresent: boolean;
}

export interface InspectorRuntimeBinding {
  kind: string;
  version?: string | null;
  path?: string | null;
  active: boolean;
  available: boolean;
  details?: string | null;
}

export interface InspectorRuntimeSnapshot {
  server: InspectorRuntimeBinding;
  php: InspectorRuntimeBinding;
  mysql?: InspectorRuntimeBinding | null;
  issues: string[];
}

export interface ReliabilityInspectorSnapshot {
  project: Project;
  diagnostics: DiagnosticItem[];
  services: ServiceState[];
  config: InspectorConfigSnapshot;
  runtime: InspectorRuntimeSnapshot;
  quickTunnel?: ProjectTunnelState | null;
  persistentHostname?: ProjectPersistentHostname | null;
  persistentTunnel?: ProjectPersistentTunnelState | null;
  persistentHealth?: PersistentTunnelHealthReport | null;
  generatedAt: string;
}

export interface ReliabilityTransferResult {
  success: true;
  path: string;
}

File: src-tauri/src/models/reliability.rs
use crate::models::diagnostics::DiagnosticItem;
use crate::models::persistent_tunnel::{
    PersistentTunnelHealthReport, ProjectPersistentHostname, ProjectPersistentTunnelState,
};
use crate::models::project::{Project, ServerType};
use crate::models::service::ServiceState;
use crate::models::tunnel::ProjectTunnelState;
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub enum ReliabilityLayer {
    Project,
    Runtime,
    Config,
    Service,
    Dns,
    Tunnel,
    Workspace,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub enum ReliabilityStatus {
    Ok,
    Warning,
    Error,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub enum ReliabilityAction {
    ProvisionProject,
    PublishPersistentDomain,
    StartProjectRuntime,
    RestoreAppMetadata,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub enum RepairWorkflow {
    Project,
    Tunnel,
    RuntimeLinks,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ReliabilityCheck {
    pub code: String,
    pub layer: ReliabilityLayer,
    pub status: ReliabilityStatus,
    pub blocking: bool,
    pub title: String,
    pub message: String,
    pub suggestion: Option<String>,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ActionPreflightReport {
    pub action: ReliabilityAction,
    pub project_id: Option<String>,
    pub ready: bool,
    pub summary: String,
    pub checks: Vec<ReliabilityCheck>,
    pub generated_at: String,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct RepairWorkflowInfo {
    pub workflow: RepairWorkflow,
    pub title: String,
    pub summary: String,
    pub touches: Vec<String>,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct RepairExecutionResult {
    pub workflow: RepairWorkflow,
    pub success: bool,
    pub message: String,
    pub touched_layers: Vec<ReliabilityLayer>,
    pub generated_at: String,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct InspectorConfigSnapshot {
    pub server_type: ServerType,
    pub output_path: String,
    pub preview: Option<String>,
    pub local_domain_alias_present: bool,
    pub persistent_alias_present: bool,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct InspectorRuntimeBinding {
    pub kind: String,
    pub version: Option<String>,
    pub path: Option<String>,
    pub active: bool,
    pub available: bool,
    pub details: Option<String>,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct InspectorRuntimeSnapshot {
    pub server: InspectorRuntimeBinding,
    pub php: InspectorRuntimeBinding,
    pub mysql: Option<InspectorRuntimeBinding>,
    pub issues: Vec<String>,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ReliabilityInspectorSnapshot {
    pub project: Project,
    pub diagnostics: Vec<DiagnosticItem>,
    pub services: Vec<ServiceState>,
    pub config: InspectorConfigSnapshot,
    pub runtime: InspectorRuntimeSnapshot,
    pub quick_tunnel: Option<ProjectTunnelState>,
    pub persistent_hostname: Option<ProjectPersistentHostname>,
    pub persistent_tunnel: Option<ProjectPersistentTunnelState>,
    pub persistent_health: Option<PersistentTunnelHealthReport>,
    pub generated_at: String,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ReliabilityTransferResult {
    pub success: bool,
    pub path: String,
}
